### PR TITLE
[WIP] first pass of destination page

### PR DESCRIPTION
**Pull request by tkell**
A shot at #674 – comments & corrections welcome!

### CODE REVIEW

**Comment by jillh510** [comment]
When there are no carpools for the destination, the page stays on "Loading carpools for this destination...." indefinitely.

**Comment by tkell** [comment]
Hmm, @jillh510, I get taken to `No carpools nearby.` when there are no carpools?  I assume you get that consistently?

**Comment by jillh510** [comment]
I get stuck on "Loading carpools for this destination...." even if there are carpools. Huh.

**Comment by TheUnlocked** [comment]
When it gets stuck on "Loading carpools for this destination....", the console emits the errors:
```
Uncaught SyntaxError: Identifier 'markers' has already been declared
    at script.js:1
```
and
```
Uncaught ReferenceError: map is not defined
    at doSearch (find.js:105)
    at setLatLng (find.js:36)
    at geoSuccess (find.js:97)
```
in that order. When it successfully shows the carpools, it still emits the first error, but it otherwise works.

**Comment by TheUnlocked** [comment]
It looks like a race condition. I'm not confident enough with git to PR onto a PR, so here are my edits to make it work:

Top of the file:
```js
let map;
// Other declarations
```

Replace `geoSuccess(position)` definition with
```js
function geoSuccess(position) {
    // zoom to user position if/when they allow location access
    if (!map){
        // wait for the map to load before proceeding
        setTimeout(() => geoSuccess(position), 100);
        return;
    }
    const coords = position.coords;
    setLatLng(coords.latitude, coords.longitude);
    zoomMap();
}
```

**Comment by jillh510** [comment]
In my version setLatLng is never getting called, which means doSearch never gets called. I've blocked the app from figuring out my location, in case that makes a difference.

**Comment by jillh510** [comment]
Wondering if the problem is here: 
if (!userQuery && !userLatLon.lat && navigator.geolocation) {
    navigator.geolocation.getCurrentPosition(geoSuccess);
}

navigator.geolocation.getCurrentPosition seems to get called, but geoSuccess never does. Maybe we need an error handler for the case where the user has denied access to their location.

**Comment by jillh510** [comment]
If I change it to '    navigator.geolocation.getCurrentPosition(geoSuccess, function () {alert('foo');});', I do see the foo alert

**Comment by tkell** [comment]
Ah, so it turns out I was doing things totally incorrectly.  I've now got the JS to a better place, but I am not sure how to generate the `geoJSONURL` that the `doSearch` function needs.  Will keep starting at it, but hopefully someone knows how to get that data.

**Comment by tkell** [comment]
OK @jillh510, I think I have the page working properly!  Do we want anymore links, or a marker to indicate the destination itself?

File: app/static/js/find.js [hunk 1]
```diff
@@ -25,9 +26,11 @@ let markers = [];
       userQuery - sanitized user query string
 */
 
-// zoom to user location only if no query in URL
-if (!userQuery && !userLatLon.lat && navigator.geolocation) {
-    navigator.geolocation.getCurrentPosition(geoSuccess);
+// zoom to user location only if no query in URL and if we're not on a Destination page
+if (!userQuery && !userLatLon.lat && !isDestinationPage && navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(geoSuccess, function() {
+		alert('You did not grant permission for the app to access your location, so we cannot show you your location.')
```
Comment (jillh510) [inline app/static/js/find.js]: Please change to "No location specified". (the user can also specify a location using the search function)

**Comment by jillh510** [comment]
I'm curious why we're using the code in find.js at all, given that the template already has the list of current carpools headed to that destination.

**Comment by dryan** [comment]
Hey @tkell, since we've got folks starting to use this this weekend, can we punt on the map for now. If we can get the list of starting points to look like the list on /carpools/find we'll be in good shape for now.

screenshot 2018-10-12 18 44 55

**Comment by tkell** [comment]
@jillh510, this is an excellent question!  I will see if I can simplify things.

@dryan, sorry, I did not see that this was supposed to be done by this weekend - and I am not sure what you mean by punting / not working on the map?

Will work on this more tomorrow evening, EST – feel free to hack on it as needed!

**Comment by dryan** [comment]
No problem. What I’m saying is if we can just get a list of places that people can leave to get to a destination that’s enough for now.

**Comment by tkell** [comment]
@jillh510, the JS does a ton of formatting that does not appear to be easily accessible from within the `destination` template, so I think I'd like to stick with it?

**Comment by tkell** [comment]
OK, I think this is working!

The JS is not great – going to see if I can split the find.js code out into another file.

File: app/static/js/find.js (hunk 1 above)
Comment (tkell) [reply app/static/js/find.js]: +1 !

File: app/static/js/find.js [hunk 2]
```diff
@@ -238,6 +240,12 @@ function showNoQuery() {
         '</div>');
 }
 
+function showDestinationQuery() {
```
[the remark this answers is not in the document]
Comment (tkell) [reply app/static/js/find.js]: Removed from the new js!

**Comment by tkell** [comment]
k, @jillh510 and @dryan, please let me know what you think!

**Comment by tkell** [comment]
Closing as we're going to us #730 !

**Pull request closed without merging**